src/rosdistro/manifest_provider/cache.py

```diff
@@ -42,19 +79,59 @@ def __init__(self, distribution_cache, manifest_providers=None):
 
     def __call__(self, dist_name, repo, pkg_name):
         assert repo.version
-        if pkg_name not in self._distribution_cache.release_package_xmls:
+        package_xml = self._distribution_cache.release_package_xmls.get(pkg_name, None)
+        if package_xml:
+            package_xml = sanitize_xml(package_xml)
+            self._distribution_cache.release_package_xmls[pkg_name] = package_xml
+            logger.debug('Loading package.xml for package "%s" from cache' % pkg_name)
+        else:
             # use manifest providers to lazy load
-            package_xml = None
             for mp in self._manifest_providers or []:
                 try:
-                    package_xml = mp(dist_name, repo, pkg_name)
+                    package_xml = sanitize_xml(mp(dist_name, repo, pkg_name))
                     break
                 except Exception as e:
                     # pass and try next manifest provider
                     logger.debug('Skipped "%s()": %s' % (mp.__name__, e))
             if package_xml is None:
                 return None
+            # populate the cache
             self._distribution_cache.release_package_xmls[pkg_name] = package_xml
+        return package_xml
+
+
+class CachedSourceManifestProvider(object):
+
+    def __init__(self, distribution_cache, source_manifest_providers=None):
+        self._distribution_cache = distribution_cache
+        self._source_manifest_providers = source_manifest_providers
+
+    def __call__(self, repo):
+        assert repo.url and repo.version
```

Comment:
SVN repositories don't have a version. Does this work or is this checking something else?